internal/web/components/pkg_overview_btn/controller.go

```diff
@@ -28,23 +28,25 @@ func getButtonId(pkgName string) string {
 func Render(w io.Writer, tmpl *template.Template, pkg *v1alpha1.Package, status *client.PackageStatus, manifest *v1alpha1.PackageManifest, latestVersion string) error {
 	buttonId := getButtonId(pkg.Name)
 	return tmpl.ExecuteTemplate(w, TemplateId, &pkgOverviewBtnInput{
-		ButtonId:        buttonId,
-		Swap:            fmt.Sprintf("outerHTML:#%s", buttonId),
-		PackageName:     pkg.Name,
-		Status:          status,
-		Manifest:        manifest,
+		ButtonId:    buttonId,
+		Swap:        fmt.Sprintf("outerHTML:#%s", buttonId),
+		PackageName: pkg.Name,
+		Status:      status,
+		Manifest:    manifest,
+		// TODO: Use semver check
```

Comment:
Same in the `pkg_detail_btns` – I think instead of comparing the semver versions, we should use the `s.updateAvailable` logic in these places too. At broadcasting time we now calculate it already anyway, so for the other two components this could be passed as parameter in the `Render` functions.

When rendering the package overview page, we also have it available already (only globally and not per package, but that's almost there I would say). It would only need to be called when rendering the package detail page. 

But it's fine to do this in a separate PR/issue. 

Also related, since we are now calling the updater more often (and with the above suggestion even more often), we should cache at least the package index calls to the repo too (`defaultClient.FetchPackageRepoIndex`). It doesn't have to be that sophisticated, I think something simple like it already is done in `defaultClient.GetLatestVersion` would be enough for now. We have a ticket for a general approach #291.

Reply:
Good idea. I did both (also added repo caching)